packages/common/src/utils/test-utils.ts

```diff
@@ -73,16 +73,18 @@ export class TestUtils {
     predicate: (state: StreamState) => boolean,
     onFailure: (state: StreamState) => void
   ): Promise<void> {
-    if (predicate(stream.state)) return
-    const timeoutPromise = new Promise((resolve) => setTimeout(resolve, timeoutMs))
-    // We do not expect this promise to return anything, so set `defaultValue` to `undefined`
-    const completionPromise = lastValueFrom(stream.pipe(filter((state) => predicate(state))), {
```

Comment:
as part of testing I noticed that `waitForState` was always taking as long as `timeoutMs` to complete, even when it completes successfully.  It turns out that the `completionPromise` here never fires.  Ideally it would be nice to fix this to use the rxjs machinery again, but I couldn't figure out how to do that, so I turned it to a simple polling-based approach that I knew would work.  This managed to significantly speed up several tests.

Reply:
Looks like the timeout in `timeoutPromise` is never cleared so it's expected it keeps the process active.
You can probably get the behavior you want using RxJS's `timeout` operator instead, ex:
```ts
const completionPromise = lastValueFrom(stream.pipe(timeout(timeoutMs), filter((state) => predicate(state))), {
  defaultValue: undefined,
})
```

Reply:
that doesn't fix it, it just makes it always throw a timeout error.  Fundamentally the core logic of `completionPromise` is just never returning.  It's not the timeoutPromise keeping things around, if I remove the timeout functionality entirely then this function just hangs forever.